jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/vsdm/ReqResForwarder.java

```diff
@@ -0,0 +1,147 @@
+package us.dot.its.jpo.ode.vsdm;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.util.Arrays;
+
+import org.apache.commons.codec.binary.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.oss.asn1.AbstractData;
+import com.oss.asn1.Coder;
+import com.oss.asn1.EncodeFailedException;
+import com.oss.asn1.EncodeNotSupportedException;
+
+import us.dot.its.jpo.ode.OdeProperties;
+import us.dot.its.jpo.ode.asn1.j2735.J2735Util;
+import us.dot.its.jpo.ode.j2735.J2735;
+import us.dot.its.jpo.ode.j2735.semi.ConnectionPoint;
+import us.dot.its.jpo.ode.j2735.semi.IPv4Address;
+import us.dot.its.jpo.ode.j2735.semi.IpAddress;
+import us.dot.its.jpo.ode.j2735.semi.PortNumber;
+import us.dot.its.jpo.ode.j2735.semi.ServiceRequest;
+import us.dot.its.jpo.ode.j2735.semi.ServiceResponse;
+
+/*
+ * This class receives service request from the OBU and forwards it to the SDC.
+ * It also receives service response from SDC and forwards it back to the OBU.
+ */
+public class ReqResForwarder implements Runnable {
+	private OdeProperties odeProps;
+	private static Coder coder = J2735.getPERUnalignedCoder();
+	private Logger logger = LoggerFactory.getLogger(this.getClass());
+	private DatagramSocket socket = null;
+	private byte[] payload;
+	private String obuReturnAddr;
+	private int obuReturnPort;
+
+	public ReqResForwarder(OdeProperties odeProps, ServiceRequest request, String obuIp, int obuPort) {
+		this.odeProps = odeProps;
+		this.obuReturnAddr = obuIp;
+		this.obuReturnPort = obuPort;
+		this.payload = createRequest(request);
+		try {
+			socket = new DatagramSocket(odeProps.getForwarderPort());
+			logger.info("ODE: Created depositor Socket with port " + odeProps.getForwarderPort());
```

Comment:
The log message doesn't need to say ODE, does it. What else is outputting to ODE log file? Fix this throughout.

Also use {} for log message arguments instead of string concatenation.